### PR TITLE
Add Term Entry: Python String Method `.join()`

### LINKED ISSUE #555: [Term Entry] Python: Strings: `.join()`
### Reason/inspiration (optional)

For the "Python: Strings" concept, write a term entry that defines the `.join()` string method, including:

- "## Syntax" and "## Example" sections with code snippets.
- At least one `codebyte/py` example (but only one) that is placed as the last example at the bottom of the term entry.

### Entry Type (select all that apply)

New Term for Existing Concept Entry

### Code of Conduct

- [X] I agree to follow Codecademy Doc's Code of Conduct

### For Maintainers

- [X] Labels added
- [x] Issue is assigned

### PR DESCRIPTION
**Pull request by KTom101**
Adding a new term entry for Python string method `.join()`, closes #555 

### Type of Change

- Adding a new entry
- Editing an existing entry (fixing a typo, bug, issues, etc)
- Updating the documentation

### Checklist

- [x] All writings are my own.
- [x] My entry follows the Codecademy Docs style guide.
- [x] My changes generate no new warnings.
- [x] I have performed a self-review of my own writing and code.
- [x] I have checked my entry and corrected any misspellings.
- [x] I have made corresponding changes to the documentation if needed.
- [x] I have confirmed my changes are not being pushed from my forked `main` branch.
- [x] I have confirmed that I'm pushing from a new branch named after the changes I'm making.

### CODE REVIEW

File: content/python/concepts/strings/terms/join/join.md
Comment (Dusch4593) [inline content/python/concepts/strings/terms/join/join.md:50]:
```suggestion
```

File: content/python/concepts/strings/terms/join/join.md
Comment (Dusch4593) [inline content/python/concepts/strings/terms/join/join.md:29]:
```suggestion
The following example returns a string version of the `fruits` list, separated by a single space (`" "`):
```

**Review by Dusch4593 — CHANGES_REQUESTED** [review]
@KTom101 Left a few comments/suggestions but, overall, this is already pretty solid! 😄

**Comment by KTom101** [comment]
@Dusch4593 Thanks, I've made the changes.

**Review by Dusch4593 — APPROVED** [review]
LGTM for the next review! 👍🏻

File: content/python/concepts/strings/terms/join/join.md
Comment (yangc95) [inline content/python/concepts/strings/terms/join/join.md:24]:
```suggestion
- The `.join()` method is called on a separator of type string. The separator can be any string, including an empty string, and is used to separate each element of the iterable.
```

I think it's important to make some distinction that the `string` mentioned in the pseudo block represents any string. This is why I'm suggesting taking away the backticks.

File: content/python/concepts/strings/terms/join/join.md
Comment (yangc95) [inline content/python/concepts/strings/terms/join/join.md:25]:
```suggestion
- The `iterable` is any object that can be iterated over like tuples or lists. All values of the `iterable` must be strings.
```

**Review by yangc95 — COMMENTED** [review]
@KTom101 Finished my review ☺️ Looks good to merge after imo 👍 @Dusch4593

**Comment by KTom101** [comment]
@yangc95 Thank you, I'm finished with the change.

**Comment by KTom101** [comment]
@yangc95 @Dusch4593  Before this is merged, do you see a good place to add the keyword "append"? This is from the shortlist where "python string append" is the keyword.

File: content/python/concepts/strings/terms/join/join.md
Comment (Dusch4593) [inline content/python/concepts/strings/terms/join/join.md:21]:
```suggestion
separator.join(iterable)
```

File: content/python/concepts/strings/terms/join/join.md
Comment (Dusch4593) [inline content/python/concepts/strings/terms/join/join.md:19]:
```suggestion

The `.join()` method is called on by being appended to a `separator` string:

```

File: content/python/concepts/strings/terms/join/join.md
Comment (Dusch4593) [inline content/python/concepts/strings/terms/join/join.md:24]:
```suggestion
- The `separator` can be any string, even an empty one, and is placed between each element from the `iterable`.
```

**Review by Dusch4593 — COMMENTED** [review]
@KTom101 Just left a few last-minute suggestions that includes inserting "append" in the entry. 😄

File: content/python/concepts/strings/terms/join/join.md
Comment (yangc95) [reply content/python/concepts/strings/terms/join/join.md:24]: This makes more sense! One bullet point for the separator and a second point for the iterable 👍 agreed

File: content/python/concepts/strings/terms/join/join.md
Comment (KTom101) [reply content/python/concepts/strings/terms/join/join.md:19]: @Dusch4593 I like this sentence but I feel like "appended" here is forced. Can we just write it as, "The `.join()` method is called on a `separator` string:" and find another place for "append".

Comment (KTom101) [reply content/python/concepts/strings/terms/join/join.md:19]: @Dusch4593 I like this sentence but I feel like "appended" here is forced. Can we just write it as, "The `.join()` method is called on a `separator` string:" and find another place for "append".

File: content/python/concepts/strings/terms/join/join.md
Comment (Dusch4593) [reply content/python/concepts/strings/terms/join/join.md:24]: So change `separator` and `iterable` to just separator and iterable?

Comment (yangc95) [reply content/python/concepts/strings/terms/join/join.md:24]: This is very "nitty-gritty."

Notice how [.replace()](https://www.codecademy.com/resources/docs/python/strings/replace) and [.index()](https://www.codecademy.com/resources/docs/python/strings/index) wrap the argument placeholders in backticks, but for [.partition()](https://www.codecademy.com/resources/docs/python/strings/partition), the keyword is not wrapped in backticks.

A similar issue in [PR 537 file](https://github.com/Codecademy/docs/pull/537/files).

So maybe for these sentence/paragraph list items, we'll do just as you said which is to change `separator` and `iterable` to separator and iterable

Comment (KTom101) [reply content/python/concepts/strings/terms/join/join.md:24]: @yangc95 maybe down the line we can have a consensus on this. I personally like using backticks around these fields only because we don't get to use many emphasis features like italics or bold.

Comment (KTom101) [reply content/python/concepts/strings/terms/join/join.md:24]: @yangc95 @Dusch4593 But to be clear, we are removing the backticks, right?

Comment (yangc95) [reply content/python/concepts/strings/terms/join/join.md:24]: Sorry, I keep going back and forth. Until there's a consensus on this, let's keep the backticks for everything except "string"

Comment (KTom101) [reply content/python/concepts/strings/terms/join/join.md:24]: ok 😂. I'll just commit everything so we can see how it would look. We can always change it again later.

**Comment by KTom101** [comment]
@Dusch4593 @yangc95 All changes have been committed. Let me know if we want to change anything else. Thanks

File: content/python/concepts/strings/terms/join/join.md
Comment (yangc95) [reply content/python/concepts/strings/terms/join/join.md]: 🥲 😆 Sounds good

**Review by yangc95 — APPROVED** [review]
Looks good for merge!

File: content/python/concepts/strings/terms/join/join.md
Comment (Dusch4593) [inline content/python/concepts/strings/terms/join/join.md:7]: Should we could part with one of these and keep two for the `Subjects`?

Comment (KTom101) [reply content/python/concepts/strings/terms/join/join.md:7]: @Dusch4593  Which one do you want to remove? I'm fine with any.

Comment (yangc95) [reply content/python/concepts/strings/terms/join/join.md:7]: @Dusch4593 were you thinking of parting with 'Code Foundations'?

Comment (Dusch4593) [reply content/python/concepts/strings/terms/join/join.md:7]: I was actually thinking maybe "Computer Science" since Python is more data science-y (right?) and strings are sort of related to code foundations(?)

Comment (KTom101) [reply content/python/concepts/strings/terms/join/join.md:7]: So I took a quick look, currently, the subjects listed for [Python Strings](https://www.codecademy.com/resources/docs/python/strings) are 'Code Foundations', 'Computer Science', 'Data Science'. 

Every string term entry, however, have only  'Data Science' & 'Computer Science' listed

Comment (Dusch4593) [reply content/python/concepts/strings/terms/join/join.md:7]: Ah, well if that's the case then yeah we should do away with "Code Foundations". 😄

**Comment by Dusch4593** [comment]
@KTom101 Woot! 🎉  The entry should now be live: https://www.codecademy.com/resources/docs/python/strings/join

**Pull request merged**
